subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c

```diff
@@ -119,13 +119,15 @@ os_mgmt_taskstat_read(struct mgmt_ctxt *ctxt)
 
 	/* Iterate the list of tasks, encoding each. */
 	for (task_idx = 0; ; task_idx++) {
+#ifdef CONFIG_THREAD_MONITOR
```

Comment:
You are turning off part of code that decides to jump out of this loop (line 124-126). I think that taskstat makes no sense if `CONFIG_THREAD_MONITOR` is disabled to the issue should be solved within Kconfig where it would not be possible to select taskstat when the monitor is not compiler in.

Reply:
I've pushed a commit that uses depends on.

Unfortunately OS_MGMT_TASKSTAT is default y.

So I can't check every enabler of OS_MGMT_TASKSTAT and have them also enable THREAD_MONITOR. 

It's not clear to me if select or depends is the best strategy here.

Reply:
I think that this is a configuration issue when an option of which whole functionality depends on completely optional, and by default not included, features of a system is defaulted to y.
Nothing else depend on this config, except for user cases and an outside software that would query the data, so there is no reason for the option to be defaulted to y by default.
I think that we should remove the default y, add note that users now have to select the thing in prj and make compilation optional for entire unit that serves taskstat.

Reply:
I agree.

Features should be default n.

Done.

Reply:
This ifdef should be changed to ifdef OS_MGMT_TASKSTAT.

Right?


```
#ifdef CONFIG_THREAD_MONITOR
static const struct k_thread *
zephyr_os_mgmt_task_at(int idx)
{

```

Reply:
"make compilation optional for entire unit that serves taskstat."

I've changed the ifdef to be on TASKSTAT instead of THREAD_MONITOR.

But I'm not sure what you are asking for here.

We could move the taskstat functions into their own C file if you wish.

Reply:
Yes, you are right, The "most local" configuration option available should be used and in this case the `OS_MGMT_TASKSTAT` would be one, as the inclusion should depend on mcumgr configuration.

Reply:
Can you please also add commit adding release note here https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/release-notes-3.0.rst#libraries--subsystems that now Taskstat will not be available for selection when thread monitor is disabled?

Reply:
Could you help me write the text? Then I can do the paperwork.

It's not clear to me how to phrase this, since AFAICT, it has never been possible to use TASKSTAT while THREAD_MONITOR was disabled.

Reply:
We should at least definitely have:

OS_MGMT_TASKSTAT is now disabled by default.

in the changelog

Reply:
> But I'm not sure what you are asking for here.
> 
> We could move the taskstat functions into their own C file if you wish.

You are probably referring to 

> (...)  as the inclusion should depend on mcumgr configuration.

my bad for not making this clear: I do not required you to move the code anywhere; the unfortunate misuse of "inclusion" is referring to pre-processor making the code available for further compilation as an opposite to excluding it from further processing.

Reply:
> Could you help me write the text? Then I can do the paperwork.
> 
> It's not clear to me how to phrase this, since AFAICT, it has never been possible to use TASKSTAT while THREAD_MONITOR was disabled.

OK, you are right. We do not have to make any addition to notes. Thanks for pointing that out.